lighty-modules/lighty-gnmi/lighty-gnmi-commons/src/main/java/io/lighty/modules/gnmi/commons/util/DataConverter.java

```diff
@@ -232,16 +236,35 @@ public static Optional<Module> findModuleByQName(@NonNull final QName element,
     }
 
     public static Optional<DataSchemaNode> findAugmentationDataNode(@NonNull final String element,
-                                                                    @NonNull final EffectiveModelContext context) {
+                                                                    @NonNull final EffectiveModelContext context,
+                                                                    @NonNull final YangInstanceIdentifier identifier) {
         return context.getModules()
                 .stream()
                 .flatMap(module -> module.getAugmentations().stream())
+                .filter(aug -> comparePathArgWithQnames(aug.getTargetPath().getNodeIdentifiers(),
+                        identifier.getPathArguments()))
                 .flatMap(augmentation -> augmentation.getChildNodes().stream())
                 .filter(childNode -> childNode.getQName().getLocalName().equals(element))
                 .map(DataSchemaNode.class::cast)
                 .findFirst();
     }
 
+    private static boolean comparePathArgWithQnames(final List<QName> qnames, final List<PathArgument> pathArguments) {
```

Comment:
This will not work because:
1) This method needs to find only augmented nodes. If searched element is not augmented, should return Optional.Empty().
2) Provided identifier is only parent YIID and this method searched for node with same name as provided element and same parent path as provided identifier.
Due to incorrect naming, I have changed naming for parameter in this method "identifier" -> "parentYIID"